File: apps/desktop/src/renderer/src/lib/api/types/discord.ts
/**
 * Discord channel bridge.
 *
 * Mirrors the FastAPI schemas in `services/agent-engine/app/schemas`.
 */

// ---- Discord channel ------------------------------------------------------

/** Response from `GET /discord/status`. */
export interface DiscordStatus {
  available: boolean;
  enabled: boolean;
  configured: boolean;
  state: string; // stopped | starting | running | error | disabled
  agent_id: number | null;
  allowed_count: number;
  message: string | null;
}

File: services/agent-engine/app/schemas/discord.py
"""Schemas for the Discord channel endpoints."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class DiscordStatusResponse(BaseModel):
    available: bool  # discord.py importable
    enabled: bool
    configured: bool  # token + agent + at least one allowed user
    state: str  # stopped | starting | running | error | disabled
    agent_id: Optional[int] = None
    allowed_count: int = 0
    message: Optional[str] = None
